lib/aur-search

```diff
@@ -121,13 +109,43 @@ case $query_type in
     search) format=${format-short} ;;
 esac
 
-# set filters
+# set parsers
+case $format in
+     long) parse() { parse_long  "$sort_key"; } ;;
+    short) parse() { parse_short "$sort_key"; } ;;
+esac
+
+# set output
 case $format in
-     long) parse() { json_long  "$sort_key"; } ;;
-    short) parse() { json_short "$sort_key"; } ;;
+     long) output() { output_long;  } ;;
+    short) output() { output_short; } ;;
 esac
 
+# prepare for execution
+
+trap_exit() {
+    if [[ ! -o xtrace ]]; then
+        rm -rf "$tmp"
+    fi
+}
+
+tmp=$(mktemp -dt "$argv0".XXXXXXXX)
+trap 'trap_exit' EXIT
+
+source /usr/share/makepkg/util/util.sh
+cd_safe "$tmp"
+
 # pipeline
-printf -- '%s\n' "$@" | aur rpc -t "$query_type" -b "$search_by" | parse
+for query in "$@"; do
+    printf -- '%s\n' "$query" | aur rpc -t "$query_type" -b "$search_by" | parse > query_results
+    if [[ -f query_results_combined ]]; then
+        mv query_results_combined query_results_prev
+        grep -Fxf query_results query_results_prev > query_results_combined
```

Comment:
Does this preserve the validity of the JSON arrays? E.g. things like resultcount will definitely be wrong

Reply:
Or I guess it parses the results first to TSV format, and then runs grep on it line-wise? Seems problematic if you want to expand functionality later

Reply:
Exactly - why is it problematic though? When a more fancier functionality is available, it can read the TSV values in variables and do whatever it needs to do, it will have to split the current "parse-and-print" function anyway.